.github/workflows/yocto-build-deploy.yml

```diff
@@ -121,37 +116,43 @@ on:
         required: false
         type: boolean
         default: ${{ startsWith(github.event.push.ref, 'refs/tags/v20') }} # Set this flag if the tag looks like an esr tag - this is good for 100 years or until balenaOS v20 comes out
+      # FIXME: add to GitHub environments
       aws_iam_role:
         description: "IAM role to assume for S3 permissions"
         required: false
         type: string
-        default: "arn:aws:iam::567579488761:role/balena-os-deploy-s3-access-Role-8r0SXqfzgolr" #TODO - Replace using environments
+        default: "arn:aws:iam::567579488761:role/balena-os-deploy-s3-access-Role-8r0SXqfzgolr"
```

Comment:
@klutchell , is this TODO removed because its uneccessary now? 

Is the IAM role the same for prod and staging? If not, don't we still need to derive this from the environment

Reply:
I moved the TODO up a few lines, it's still there.

We do have different IAM roles for prod / staging and they are defined [here](https://github.com/balena-os/.github/blob/cf4a83914b1009c6eb0806f2a4093897b0401df3/.github/settings.yml#L28-L31) and [here](https://github.com/balena-os/.github/blob/cf4a83914b1009c6eb0806f2a4093897b0401df3/.github/settings.yml#L80-L81).

In a future PR I will update the workflow to use env secrets where appropriate. For now this PR is big enough.